### DISCUSSION TITLE
Compiling external scanner with sanitizers and loading it from tree-sitter-cli

### DISCUSSION THREAD

**Discussion by ahelwer** [tree-sitter/tree-sitter#2076] [Q&A] [answered]
I posted a question on StackOverflow [here](https://stackoverflow.com/q/75376231/2852699) and would appreciate any help I can get on this. I wrote [a post](https://ahelwer.ca/post/2023-02-04-cpp-bugs/) on two difficult bugs I ended up writing in my external scanner, and became interested in the question of whether sanitizers could have caught them (and might catch others). So I am trying to manually compile my grammar with `-fsanitize=address,undefined` then load it from the tree-sitter CLI which is being uncooperative. Do you think this is possible or should I just write my own front-end in C++? It would be sad for that to happen since I really want to run `tree-sitter test` and don't feel like writing all that test file parsing logic in C++.

**Comment by ObserverOfTime** [comment, marked as answer]
- Option 1: Export `CFLAGS`
- Option 2: Run `tree-sitter fuzz`
- Option 3: Use the [fuzz-action](https://github.com/tree-sitter/fuzz-action) locally
